### PR TITLE
HACDOCS-588

### PR DESCRIPTION
**Pull request by Chr1st1anSears**
Adds doc for preventing redundant rebuilds w on-cel-expression

### CODE REVIEW

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_preventing_redundant_rebuilds.adoc [hunk 1]
```diff
@@ -0,0 +1,60 @@
+= Preventing redundant rebuilds
+
+After you upgrade the build pipeline for a component, {ProductName} rebuilds the component when new pull requests (PRs) include changes to that component. This behavior helps you identify any issues the PR might cause before merging it. 
+
+By default, {ProductName} configures upgraded pipelines to rebuild a component _only_ if a PR includes changes to that component. This configuration prevents redundant rebuilds in repositories that contain multiple components. 
+
+However, depending on the structure of your repository and the dependencies of your component, you may want to adjust the default configuration for rebuilds to better suit your needs. In particular, the default configuration might still rebuild a component too often. You can change this behavior by editing the `on-cel-expression` annotation.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_preventing_redundant_rebuilds.adoc:7]: This is not displaying correctly in the output -  `on-cel-expression`

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_preventing_redundant_rebuilds.adoc [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
++
+----
+pipelinesascode.tekton.dev/on-cel-expression: event == "pull_request" && target_branch
+      == "main" && ( "my-component/context/***".pathChanged() || ".tekton/my-component-pull-request.yaml".pathChanged()
+      || "root-docker/Dockerfile".pathChanged() )
+----
++
+WARNING: Misconfiguring this value can prevent builds from starting or cause them to start at the wrong time.
++
+. Commit this change to your repository.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_preventing_redundant_rebuilds.adoc:35]: This?
As there can be multiple paths, I believe we can keep it generic

```suggestion
. Commit the changes.
```

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_preventing_redundant_rebuilds.adoc [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+. Allow {ProductName} to rebuild your component. 
+
+.Verification
+To verify that you have prevented redundant rebuilds: 
+
+. Open a PR that modifies the component that you want {ProductName} to rebuild. 
+. Allow {ProductName} to rebuild your component.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_preventing_redundant_rebuilds.adoc:42]: I don't think there is a button to `Allow`.

```suggestion
. {ProductName} rebuilds your component.
```

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_preventing_redundant_rebuilds.adoc [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+----
+pipelinesascode.tekton.dev/on-cel-expression: event == "pull_request" && target_branch
+      == "main" && ( "my-component/context/***".pathChanged() || ".tekton/my-component-pull-request.yaml".pathChanged()
+      || "root-docker/Dockerfile".pathChanged() )
+----
++
+WARNING: Misconfiguring this value can prevent builds from starting or cause them to start at the wrong time.
++
+. Commit this change to your repository.
+. Allow {ProductName} to rebuild your component.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_preventing_redundant_rebuilds.adoc:36]: Ignore if not true.

```suggestion
. Wait for {ProductName} to rebuild your component. 
```

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_preventing_redundant_rebuilds.adoc [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
++
+. Commit this change to your repository.
+. Allow {ProductName} to rebuild your component. 
+
+.Verification
+To verify that you have prevented redundant rebuilds: 
+
+. Open a PR that modifies the component that you want {ProductName} to rebuild. 
+. Allow {ProductName} to rebuild your component.
+. Check the PipelineRuns page in {ProductName}. There should be only one new PipelineRun with the type “Build.” Check its logs for the `build-container` task. They should only refer to the component you specified with `on-cel-expression`.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_preventing_redundant_rebuilds.adoc:43]:
```suggestion
. Check the PipelineRuns page in {ProductName}. There should be only one new PipelineRun with the type as “Build.” Check its logs for the `build-container` task. They should only refer to the component you specified with `on-cel-expression`.
```

**Review by gtrivedi88 — APPROVED** [review]

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_preventing_redundant_rebuilds.adoc (hunk 1 above)
Comment (Chr1st1anSears) [reply docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_preventing_redundant_rebuilds.adoc:7]: Hmm for me it is correctly displaying. Is it possible there's a browser issue on your end?

**Pull request merged**